## 1. Business Understanding

## 2. Data Understanding
#### Step 1: Data Collection and Initial Exploration

In [1]:
import pandas as pd

# Load the dataset
file_path = 'data/phishing_dataset/phishing_dataset.csv'
df = pd.read_csv(file_path)

# Display basic information about the dataset
print("Dataset Shape:", df.shape)
print("\nFirst 5 Rows of the Dataset:")
display(df.head())

print("\nData Types and Null Values:")
display(df.info())


Dataset Shape: (504983, 8)

First 5 Rows of the Dataset:


url    target  url_length  \
0  https://docs.google.com/presentation/d/e/2PACX...  Phishing         178   
1    https://btttelecommunniccatiion.weeblysite.com/  Phishing          47   
2                        https://kq0hgp.webwave.dev/  Phishing          27   
3  https://brittishtele1bt-69836.getresponsesite....  Phishing          50   
4         https://bt-internet-105056.weeblysite.com/  Phishing          42   

   hostname_length  tld  num_dots  has_at_symbol  https  
0               15  com         3          False   True  
1               38  com         2          False   True  
2               18  dev         2          False   True  
3               41  com         2          False   True  
4               33  com         2          False   True


Data Types and Null Values:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 504983 entries, 0 to 504982
Data columns (total 8 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   url              504983 non-null  object
 1   target           504983 non-null  object
 2   url_length       504983 non-null  int64 
 3   hostname_length  504983 non-null  int64 
 4   tld              504755 non-null  object
 5   num_dots         504983 non-null  int64 
 6   has_at_symbol    504983 non-null  bool  
 7   https            504983 non-null  bool  
dtypes: bool(2), int64(3), object(3)
memory usage: 24.1+ MB


None

> **Key Insights:**
> There are 504,983 rows and 8 columns.
>> **Column Definitions**
>> - ***url:*** Contains URLs, likely representing the samples. This column is of type object.
>> - ***target:*** Appears to be the target variable, with entries indicating whether each URL is phishing or legitimate. This column is also of type object.
>> - ***url_length and hostname_length:*** Numerical columns (int64) likely representing the length of the full URL and the hostname, respectively.
>> - ***tld:*** Represents the top-level domain of each URL, such as ".com" or ".org". This column has a small number of missing values (504,755 non-null out of 504,983).
>> - ***num_dots:*** An integer feature that likely counts the number of dots (.) in the URL.
>> - ***has_at_symbol and https:*** Boolean columns indicating whether the URL contains an "@" symbol and whether it uses HTTPS, respectively.